index.js

```diff
@@ -85,12 +102,12 @@ Block.prototype.isGenesis = function () {
  * @method setGenesisParams
  */
 Block.prototype.setGenesisParams = function () {
-  this.header.gasLimit = params.genesisGasLimit.v
-  this.header.difficulty = params.genesisDifficulty.v
-  this.header.extraData = params.genesisExtraData.v
-  this.header.nonce = params.genesisNonce.v
-  this.header.stateRoot = params.genesisStateRoot.v
-  this.header.number = new Buffer([])
+  this.header.gasLimit = this._common.genesis().gasLimit
+  this.header.difficulty = this._common.genesis().difficulty
+  this.header.extraData = this._common.genesis().extraData
+  this.header.nonce = this._common.genesis().nonce
+  this.header.stateRoot = this._common.genesis().stateRoot
+  this.header.number = Buffer.from([0])
 }
```

Comment:
Should we move this code into a new BlockHeader.setGenesisParams() method? That way it will be consistent with how .hash() and .isGenesis() work (both forward the calls to BlockHeader)

Reply:
Yes, totally makes sense.

Reply:
This ``Buffer.from([0])`` change is correct, isn't it?

Reply:
Thanks for bringing that up. I didn't notice that change.

It's should actually be Buffer.from([]) instead of Buffer.from[0]. The RLP encoding of the genesis block contains an empty buffer for the number field instead of a buffer containing 0.

Reply:
Ok, thanks, changed this back again, wasn't sure on this one.